### PR TITLE
Circle CI configuration for SalesforceMobileSDK-ReactNative

### PR DESCRIPTION
**Pull request by wmathurin**
* generate workspace
* build workspace
* run tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  - &generate-workspace
+    name: Setup
+    command: node ./prepareios.js
+    when: always
+
+mac: &mac
+  working_directory: ~/SalesforceMobileSDK-ReactNative/ios/SalesforceReactTests
+  macos:
+    xcode: "9.4.1"
```
Comment (bhariharan) [inline .circleci/config.yml]: Shouldn't this be `Xcode 10` now?

Comment (wmathurin) [reply .circleci/config.yml]: Probably - this is just a first cut (basically a copy/paste job on https://github.com/forcedotcom/SalesforceMobileSDK-Package/blob/master/.circleci/config.yml). I just want to see it running. Then I will refine.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    BASH_ENV: ~/.bashrc
+    FASTLANE_SKIP_UPDATE_CHECK: true
+    CHRUBY_VER: 2.5.1
+
+version: 2
+jobs:
+  test-ios:
+    <<: *defaults
+    steps:
+      - checkout
```
Comment (bhariharan) [inline .circleci/config.yml]: What's under the `checkout` step here (should it be `checkout:`)?

File: .circleci/fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      xcodebuild(
+          xcargs: 'CLANG_ANALYZER_OUTPUT=plist-html CLANG_ANALYZER_OUTPUT_DIR=./clangReport RUN_CLANG_STATIC_ANALYZER=YES',
+          workspace: 'SalesforceReactTestApp.xcworkspace',
+          scheme: scheme,
+          sdk: 'iphonesimulator',
+          )
+    ensure
+      #move clangReports to one folder
+      system('mkdir -p ../clangReport/StaticAnalyzer')
+#      system('mv ../libs/SmartSync/clangReport/StaticAnalyzer/SmartSync ../clangReport/StaticAnalyzer/')
```
Comment (bhariharan) [inline .circleci/fastlane/Fastfile]: Should this commented out line be removed (it's referencing `SmartSync`)?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            popd
+
+      - run:
+          name: Creating index.ios.bundle
+          command: | 
+            pushd ios/SalesforceReactTests
+            node ./updatebundle.js
+            popd
+      
+#      - run:
```
Comment (bhariharan) [inline .circleci/config.yml]: Should this commented out section go away?

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+version: 2
+jobs:
+  test-ios:
+    <<: *defaults
+    steps:
+      - checkout
+
+#      - restore_cache:
+#          key: v5-gems-{{ checksum ".circleci/Gemfile.lock" }}
```
Comment (bhariharan) [inline .circleci/config.yml]: Should this commented out section go away?

Comment (wmathurin) [reply .circleci/config.yml]: This is still work in progress.
When the PR is ready to merge, those commented sections should either be uncommented (and working) or gone.

File: .circleci/config.yml (hunk 2 above)
Comment (wmathurin) [reply .circleci/config.yml]: Nothing is required under checkout (see https://circleci.com/docs/2.0/configuration-reference/#checkout).

**Comment by wmathurin** [comment]
Tests are running (and passing) on CI.

**Pull request merged**